src/ParcelRegistry/Parcel/Parcel.cs

```diff
@@ -70,6 +72,38 @@ public void AttachAddress(AddressPersistentLocalId addressPersistentLocalId)
             ApplyChange(new ParcelAddressWasAttachedV2(ParcelId, addressPersistentLocalId));
         }
 
+        public void DetachAddress(AddressPersistentLocalId addressPersistentLocalId)
+        {
+            GuardParcelNotRemoved();
+
+            var address = _addresses.GetOptional(addressPersistentLocalId);
+
+            if (address is null)
+            {
+                throw new AddressNotFoundException();
+            }
+
+            if (address.Value.IsRemoved)
+            {
+                throw new AddressIsRemovedException();
+            }
```

Comment:
Shouldn't this be idempotent?
If it is removed, then it shouldn't be here in the first place so you can accept the request to detach.

And should `IsNotFound` check not be on `AddressPersistentLocalIds` ?